spec/classes/init_spec.rb

```diff
@@ -1,5 +1,13 @@
 require 'spec_helper'
 describe 'nrpe' do
+  # setting the global facts for all test if not overwritten locally
+  let(:facts) do
+    {
+      :osfamily          => 'RedHat',
+      :lsbmajdistrelease => '6',
+      :architecture      => 'x86_64',
```

Comment:
Adding :architecture does fix the issues with STRICT_VARIABLES on P4.4.
Refactored to become DRY compliant.